RecoJets/JetProducers/plugins/PileupJetIdProducer.cc

```diff
@@ -135,27 +139,27 @@ PileupJetIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 			if( jecCor_ == 0 ) {
 				initJetEnergyCorrector( iSetup, iEvent.isRealData() );
 			}
-			//if( ispat ) {
-			//	jecCor_->setJetPt(patjet->correctedJet(0).pt());
-			//} else {
-			jecCor_->setJetPt(jet.pt());
-			//}
+			if( ispat ) {
+				jecCor_->setJetPt(patjet->correctedJet(0).pt());
+			} else {
+			        jecCor_->setJetPt(jet.pt());
+			}
 			jecCor_->setJetEta(jet.eta());
 			jecCor_->setJetA(jet.jetArea());
 			jecCor_->setRho(rho);
 			jec = jecCor_->getCorrection();
 		}
 		// If it was requested AND the input is an uncorrected jet apply the JEC
-		bool applyJec = applyJec_ && !inputIsCorrected_;  //( ! ispat && ! inputIsCorrected_ );
+		bool applyJec = applyJec_ && ( ispat || !inputIsCorrected_ );
 		reco::Jet * corrJet = 0;
```

Comment:
It would be safer to use unique_ptr rather than a bare pointer here, but that is not essential.